### PR TITLE
fix(helm): Update status output to include resource details

### LINKED ISSUE #4712: Helm 2.11.0 status output differs for deployments
In helm version 2.11.0 the helm status output differs for the deployment.

In previous version (2.10.0) it showed
```
==> v1beta2/Deployment
NAME            DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
metrics-server  1        1        1           1          1d
```

In version 2.11.0 is shows
```
==> v1beta2/Deployment
NAME                           AGE
unhinged-hydra-metrics-server  20s
```

```
kubectl get deployments
NAME                            DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
unhinged-hydra-metrics-server   1         1         1            1           3m
```

So there is some information that list lost in the status output in version 2.11.0.
Is this expected behaviour?

Output of `helm version`: 
```
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"e8c8ac53ee97f1968922718551a275c0abd15c3c", GitTreeState:"clean"}
```

Output of `kubectl version`: 
```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-09-07T15:46:48Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-09-07T15:45:01Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  N/A

### PR DESCRIPTION
**Pull request by mortent**
Update of the client-go package changed the status output to only include
the age of resources. The new printer in client-go only formats the output
to include details of specific resources if the internal representation
of resources are passed into the printer. This PR updates helm to convert
resources to the internal type before printing.

Closes #4712

Signed-off-by: Morten Torkildsen <mortent@google.com>

### CODE REVIEW

**Review by bacongobbler — COMMENTED** [review]
looks good to me, but I'd love to get a second pair of eyes on this (optimally @adamreese) just to make sure I'm not missing something.

**Review by bacongobbler — APPROVED** [review]
got the LGTM from @adamreese offline. Thanks!

**Comment by ocofaigh** [comment]
Anybody know when v2.11.1 will be released?

**Pull request merged**
